common/src/main/java/com/scottlogic/deg/common/profile/ConstraintType.java

```diff
@@ -0,0 +1,69 @@
+package com.scottlogic.deg.common.profile;
+
+public enum ConstraintType
+{
+    EQUAL_TO("equalTo"),
+    IN_SET("inSet"),
+    IN_MAP("inMap"),
+    NULL("null"),
+    GRANULAR_TO("granularTo"),
+    MATCHES_REGEX("matchingRegex"),
+    CONTAINS_REGEX("containingRegex"),
+    OF_LENGTH("ofLength"),
+    LONGER_THAN("longerThan"),
+    SHORTER_THAN("shorterThan"),
+    GREATER_THAN("greaterThan"),
+    GREATER_THAN_OR_EQUAL_TO("greaterThanOrEqualTo"),
+    LESS_THAN("lessThan"),
+    LESS_THAN_OR_EQUAL_TO("lessThanOrEqualTo"),
+    AFTER("after"),
+    AFTER_OR_AT("afterOrAt"),
+    BEFORE("before"),
+    BEFORE_OR_AT("beforeOrAt"),
+    NOT("not"),
+    ANY_OF("anyOf"),
+    ALL_OF("allOf"),
+    CONDITION("condition");
+
+    private final String type;
+
+    ConstraintType(String name)
+    {
+        this.type = name;
+    }
+
+    public String getType()
+    {
+        return type;
+    }
+
+    public static ConstraintType fromString(String type){
```

Comment:
We could construct the map here by calling something like

`Arrays.stream(values()).collect(Collectors.toMap(ConstraintType::getType, Function.identity()))`

instead of maintaining this list manually.

Reply:
Yes I agree

Reply:
I have added it to the constraint-types branch which will be in the next pull request